replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/WorkerManager.java

```diff
@@ -0,0 +1,123 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.impl.worker;
+
+import com.connexta.replication.api.data.NotFoundException;
+import com.connexta.replication.api.data.ReplicationPersistenceException;
+import com.connexta.replication.api.data.Site;
+import com.connexta.replication.api.impl.NodeAdapters;
+import com.connexta.replication.api.persistence.SiteManager;
+import com.connexta.replication.api.queue.QueueBroker;
+import com.connexta.replication.spring.ReplicationProperties;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class responsible for managing item workers. It will monitor sites currently available in storage
+ * and create/destroy item workers accordingly for each site, respecting each site's parallelism
+ * factor.
+ */
+public class WorkerManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WorkerManager.class);
+
+  private final ScheduledExecutorService executorService = Executors
+      .newSingleThreadScheduledExecutor();
+
+  private final QueueBroker broker;
+
+  private final Set<String> siteIds;
+
+  private final Site localSite;
+
+  private final SiteManager siteManager;
+
+  private final NodeAdapters nodeAdapters;
+
+  private final Map<String, WorkerThreadPool> threadPools = new HashMap<>();
+
+  /**
+   * Creates a new worker manager.
+   *
+   * @param broker the queue broker for retrieving queues
+   * @param siteManager the site data access manager
+   * @param nodeAdapters node adapters factory
+   * @param properties replication properties
+   */
+  public WorkerManager(QueueBroker broker, SiteManager siteManager, NodeAdapters nodeAdapters,
+      ReplicationProperties properties) {
+    this.broker = broker;
+    this.siteManager = siteManager;
+    this.nodeAdapters = nodeAdapters;
+    this.siteIds = new HashSet<>(properties.getSites());
+    this.localSite = siteManager.get(properties.getLocalSite());
+  }
+
+  /**
+   * Initializes this manager and begins monitoring of sites.
+   */
+  public void init() {
+    executorService.scheduleAtFixedRate(this::monitorSites, 0L, 30L, TimeUnit.SECONDS);
+  }
+
+  private void monitorSites() {
```

Comment:
❗️ We should also make sure to filter any sites we don't support (not just those not configured in `sites`). The kind and the type of the site might be unknown if this is an older version of the manager.